### ISSUE TITLE
[bug] Multiple Tokens Broadcastings when Broadcast token is absent in overlapping channels

### ISSUE THREAD

**Issue by KokopeIlI** [Placeholder-Software/Dissonance#102]
## Context
Currently Using several colliders as triggers to act as private rooms. There is a public global channel that ALL players should be able to listen to at all times. and there are private rooms that no one but te individuals in the private rooms can hear. Currently, when a player moves from the public room, to the private room, the public room can still hear the private room, even when the public broadcasting channel token is absent

## Expected Behavior
A player should be able to hold a private conversation in a private room while still being able to hear audio from the public room.

## Actual Behavior
Players enter a private room, but still broadcast to overlapping public room, even without a proadcasting token

## Workaround
None found

## Fix
_Not obligatory! If you think you know exactly how this problem should be fixed tell us how_

## Steps to Reproduce
_Provide a detailed set of steps to reproduce the problem_
1. Create a private room that is activated manually using a capsule collider, not by using the triggers on the components.
2. Create a Public room with separate tokens for broadcast and receive
3. when moving from public to private remove ONLY the public broadcasting token
4. speak from private room and the players in the public room will still hear you.

## Your Environment
_Include as many relevant details about the environment you experienced the bug in_

- **Dissonance version used**: e.g. v6.2.0

- **Unity version**: e.g. 2017.3.1f1 Personal 

- **Editor Operating System and version**: Windows 10 Home

- **Build Settings**: Windows, x86_64, Development Build

**Comment by KokopeIlI** [comment]
Martindevans, I'm sorry for the late response, I have adjusted the above code to reflect your changes but I am still encountering the same issue.

**Comment by martindevans** [comment]
Could you email me (martin@placeholder-software.co.uk) your invoice number. I'll send you back the latest version of Dissonance (currently awaiting approval from Unity to release on the asset store). I fixed the bug but in a slightly different way to how I mentioned above, hopefully the temp fix I told you before was wrong and the real fix will resolve your issue :)

**Comment by KokopeIlI** [comment]
Yes of course. I have to say your turn around time is incredible. Thank you
for your dedication!

IN010000419953

**Comment by martindevans** [comment]
Woah, don't give out your invoice number in public! You'll need to email me directly (rather than reply to this issue thread) so that I know who to send back to :)

**Comment by martindevans** [comment]
Reported not fixed by latest update:

> What I did try was changing it to push to talk and it worked just fine. But I found that if I held down the PTT when I moved from a public region to a private one it would continue to broadcast. I dont know if that helps, let me know if there is anything more I can do.

**Comment by martindevans** [comment]
When you say:

> Create a private room that is activated manually using a capsule collider, not by using the triggers on the components.

What exactly are you doing to "activate" the room? Just enabling/disabling it when the capsule enters/leaves?

**Comment by KokopeIlI** [comment]
I have several private "regions"  that are static in the world, when a
player passes through them it calls a trigger method(the one that I sent to
you) that adds the Broadcast/Receive token to the comms object for the
private channel, and then removes just the "PublicChat" token. It does the
reverse when a player steps out of a region.

I would have to say that it is not an entirely accurate description since
the audio regions are static from the get go, sorry for the confusion

**Comment by martindevans** [comment]
Ah yes, I forgot about the reddit discussion with the code samples. I'll go check those out again. Thanks :)

**Comment by KokopeIlI** [comment]
Let me know if you want me to test/try anything. Im at work till 6 but I'll
be able to work on it after that.

**Comment by martindevans** [comment]
I can't reproduce this issue with the latest changes. I made some changes to the demo scene:

1. Change global chat room broadcaster to require `Test Token`
2. Change global chat room to voice activation
2. Add this code:

```
void OnGUI()
{
    if (GUILayout.Button("Add 'Test Token' To DissonanceComms"))
        FindObjectOfType<DissonanceComms>().AddToken("Test Token");
    if (GUILayout.Button("Remove 'Test Token' From DissonanceComms"))
        FindObjectOfType<DissonanceComms>().RemoveToken("Test Token");
}
```

When using Dissonance 6.2.3 (the one I sent to you by email) this works. Speaking continuously (so it never stops broadcasting) and then clicking "Remove" cuts off the broadcast. If I try this same test with Dissonance 6.2.0 clicking "Remove" does _not_ cut me off (until I stop speaking).

If you try this with a demo scene using Dissonance 6.2.3 does it work for you? Do you agree that this is a good approximation of the problem you're having?

**Comment by martindevans** [comment]
Dissonance 6.2.4 went live on the asset store yesterday. As far as I can tell this was fixed in the previous version (6.2.3) so I'll close this issue now - of course please re-open it if the issue persists.
